endor/src/services/invites.service.js

```diff
@@ -0,0 +1,187 @@
+   *   as a new invite or as an update
+   * @returns {Array} list of errors
+   */
+  static async validateInvite(invite, asNewInvite) {
+    const errors = [];
+    if (asNewInvite) {
+      if (!('userInvitedId' in invite)) {
+        errors.push(new RequestParamError('userInvitedId', 'User is required.'));
+      }
+
+      if (!('projectInvitedToId' in invite)) {
+        errors.push(new RequestParamError('projectInvitedToId', 'Project is required.'));
+      }
+    } else if (!('status' in invite)) {
+      errors.push(new RequestParamError('status', 'Status is required for invite updates.'));
+    }
+
+    if (invite.daysFromCreationUntilExpiration
+      && !isNonNegativeInteger(invite.daysFromCreationUntilExpiration)) {
+      errors.push(new RequestParamError('daysFromCreationUntilExpiration', 'Must be a non-negative integer.'));
+    }
+
+    if (invite.status && !InviteService.isValidInviteStatus(invite.status)) {
+      const inviteStatusValues = Object.keys(InviteStatus).map(key => InviteStatus[key]);
+      const enumValues = inviteStatusValues.join(', ');
+      errors.push(new RequestParamError('status', `Must be one of the following: ${enumValues}`));
+    }
+
+    return errors;
+  }
+
+  /**
+   * Gets a invite by the invite id
+   * @param inviteId the invite id to find by
+   * @returns {Object} the invite that was found
+   */
+  async getInviteById(inviteId) {
+    this.log.info(`InviteService: find invite with id of ${inviteId}`);
+    const inviteFound = await this.inviteRepository.findOne({
+      where: {
+        id: inviteId
+      }
+    });
+
+    if (inviteFound === null) {
+      throw new InviteNotFoundException(`Invite with ${inviteId} could not be found.`);
+    }
+
+    return inviteFound;
+  }
+
+  /**
+   * Gets all invites for the given project
+   * @param projectId the project id to find by
+   * @returns {Object} the invite that was found
+   */
+  async getInvitesByProjectId(projectId) {
+    this.log.info(`InviteService: find invite with project id of ${projectId}`);
+    const invitesFound = await this.inviteRepository.findAll({
+      where: {
+        projectInvitedToId: projectId
+      }
+    });
+
+    if (invitesFound === null) {
+      throw new InviteNotFoundException(`Invites belonging to project ${projectId} could not be found.`);
+    }
+
+    return invitesFound;
+  }
+
+  /**
+   * Gets all invites for the given user
+   * @param userId the user id to find by
+   * @returns {Object} the invite that was found
+   */
+  async getInvitesByUserId(userId) {
+    this.log.info(`InviteService: find invite with user id of ${userId}`);
+    const invitesFound = await this.inviteRepository.findAll({
+      where: {
+        userInvitedId: userId
+      }
+    });
+
+    if (invitesFound === null) {
+      throw new InviteNotFoundException(`Invites for user ${userId} could not be found.`);
+    }
+
+    return invitesFound;
+  }
```

Comment:
Should there be a 404 for user not found here? What would happen if the user doesn't exist?